src/wp-includes/link-template.php

```diff
@@ -1977,7 +1977,7 @@ function get_adjacent_post( $in_same_term = false, $excluded_terms = '', $previo
 	$sort = apply_filters( "get_{$adjacent}_post_sort", "ORDER BY p.post_date $order LIMIT 1", $post, $order );
 
 	$query     = "SELECT p.ID FROM $wpdb->posts AS p $join $where $sort";
-	$query_key = 'adjacent_post_' . md5( $query );
+	$query_key = wp_cache_get_last_changed( 'posts' ) . '_adjacent_post_' . md5( $query );
```

Comment:
This is been worked on in PR https://github.com/WordPress/wordpress-develop/pull/3366

Reply:
Reverted it from this PR in preference for that PR.